PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStorageCtrl.kt

```diff
@@ -265,8 +266,9 @@ class SparkSubmissionJobUploadStorageCtrl(val view: SparkSubmissionJobUploadStor
 
         val rawStoragePath = "$schema://$container@$fullStorageBlobName"
         return if (schema.startsWith(ADLSGen2StorageAccount.DefaultScheme))
-            "${ADLSGen2FSOperation.converToGen2Path(URI.create(rawStoragePath))}/${SparkSubmissionContentPanel.Constants.submissionFolder}/"
-        else  "$rawStoragePath/${SparkSubmissionContentPanel.Constants.submissionFolder}/"
+            AbfsUri.parse("$rawStoragePath/").url
+                .resolve("${SparkSubmissionContentPanel.Constants.submissionFolder}/").toString()
+        else "$rawStoragePath/${SparkSubmissionContentPanel.Constants.submissionFolder}/"
```

Comment:
Have parsed above, do we still need this else?

Reply:
Use `else` statement to handle non-Gen2 storage path

---
In reply to: [361566196](https://github.com/microsoft/azure-tools-for-java/pull/3885#discussion_r361566196) [](ancestors = 361566196)